sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -289,6 +289,12 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case ShowCreateTable(_: ResolvedTable, _) =>
       throw new AnalysisException("SHOW CREATE TABLE is not supported for v2 tables.")
 
+    case CacheTable(_: ResolvedTable, _, _, _) =>
+      throw new AnalysisException("CACHE TABLE is not supported for v2 tables.")
```

Comment:
We don't need new v2 APIs to support it. This command touches `CacheManager` which is Spark internal.

Reply:
Ah OK. An existing bug, I guess? (it only supported temp view / v1 tables).

Does it make sense to match `case CacheTable(_: ResolvedTable, _, _, _) ` in `ResolveSessionCatalog` or should we match it in `DataSourceV2Strategy` with a new `CacheTableExec` similar to `CacheTableCommand`?

Reply:
We can add a `CacheTableExec` as a v2 version of `CacheTableCommand`.

Reply:
Added v2 version.